docs: explain rootless limitation and constraint (#5809)
dagger/dagger: docs/current/faq.md

Choosing the right configuration depends on the kernel capabilities, its own configuration, fuse availability and the upstream kernel patches specific to the host distribution.
Meaning that if it's not possible to use the default snapshotter, you will experience noticeable slowdown, either with fuse-overlayfs or the default snapshotter.

Dagger aims to accelerate your CI/CD pipeline, reducing performances also reduces the value Dagger adds as well as leading to unstable behaviour.

#### Network constraint

Using a rootless engine will especially constraint the custom network management of Dagger.

It is possible to use [slirp](https://github.com/rootless-containers/slirp4netns) as a workaround, but it will highly decrease network performances in addition to be extremely complex to setup.
Rootlesskit wrote a [table of comparaison](https://github.com/rootless-containers/rootlesskit/blob/master/docs/network.md#network-drivers), we can observe that `slirp` is at least five time slower than root network, meaning that dagger will be way slower if ran as rootless.

#### Conclusion

> It might be possible in the future once these constraint vanish but for now it's not possible to run dagger engine with privileges if you want to unlock its full potentiel.

Reviewer: >potentiel.

nit: french spelling of potential.

> It might be possible in the future once these constraint vanish but for now it's highly recommended to run dagger engine with privileges if you want to unlock its full potentiel.

Reviewer 2: I think this should be stronger - we don't just recommend running the dagger engine with privileges, it's not really possible to do so without.

Author: I said highly recommended in case some people really want to, it might be possible but we will not support that decision, I couldn't find a way when I tried but maybe someone with more experience with rootless tech can, I do not know.

I can write it is not possible if you think it's totally impossible

Reviewer 2: I doubt it's totally impossible - but it will require substantial code changes in the engine IMO.

I just want to make sure users don't go away and spend time trying to get it working, if they think it can be done with just a few surface level changes, since that's likely to make people unhappy :cry:

Author: Alright! 🚀  Fixed in [a901bc6](https://github.com/dagger/dagger/pull/5809/commits/a901bc6a0cd7a319f0dfed3df420ac112731917e)

Operations are automatically cached every time a Dagger pipeline runs. [Cache volumes](./quickstart/635927-caching.mdx) must be explicity defined and used in your Dagger pipeline code.

### Can I run the Dagger Engine as a "rootless" container?

> No. "Rootless mode" means running the Dagger Engine as a container without the `--privileged` flag. In this case, the container would not have "root" privileges for all devices on the host system. Currently, the Dagger Engine cannot be run as a rootless container; network, platform and volume constraints related to rootless usage would significantly limit its capabilities and performance. [Read more about these constraints](https://github.com/dagger/dagger/blob/main/core/docs/d7yxc-operator_manual.md).

Reviewer 3: ```suggestion
No. "Rootless mode" means running the Dagger Engine as a container without the `--privileged` flag. In this case, the container would not run as the "root" user of the system. Currently, the Dagger Engine cannot be run as a rootless container; network and filesystem constraints related to rootless usage would significantly limit its capabilities and performance. [Read more about these constraints](https://github.com/dagger/dagger/blob/main/core/docs/d7yxc-operator_manual.md).
```
